packages/client/src/game-core/input/InputHandler.ts

```diff
@@ -0,0 +1,97 @@
+import { Player, State } from '../Player'
+import { Indexed } from '../utils/common-types'
+
+export class InputHandler {
+  private player: Player
+  private pressedKeys: Indexed<boolean> = {}
+
+  constructor(obj: Player) {
+    this.player = obj
+
+    document.addEventListener('keydown', e => this.setKey(e, true))
+
+    document.addEventListener('keyup', e => this.setKey(e, false))
+
+    window.addEventListener('blur', e => {
+      this.pressedKeys = {}
+    })
+  }
+
+  private setKey(e: KeyboardEvent, pressed: boolean) {
+    const code = e.code
+    let key: string
```

Comment:
`switch case` лучше вынести в отдельную функцию с возвратом ключа и сразу присваивать в `key`.
```suggestion
    const key: string = getKeyByCode(code);
```